apis/secretsmanager/v1alpha1/secret_types.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// Keys used in connection secret.
+const (
+	ConnectionSecretKeyName        = "secret"
+	ConnectionSecretKeyProjectName = "projectName"
+)
+
+// SecretParameters defines parameters for a desired Secret Manager's secret.
+type SecretParameters struct {
+
+	// Required. The resource name of the project to associate with the
+	// [Secret][google.cloud.secretmanager.v1.Secret], in the format `projects/*`.
+	Parent string `json:"parent,omitempty"`
```

Comment:
If required we don't need `omitempty`.
```suggestion
	Parent string `json:"parent"`
```

Reply:
This should also be a cross resource reference to the `Secret` type.